fuels-abigen-macro/tests/test_projects/foo-caller-contract/src/main.sw

```diff
@@ -0,0 +1,16 @@
+contract;
+
+use foo::FooContract;
+use std::constants::ETH_ID;
+
+abi FooCaller {
+    fn call_foo_contract(gas_: u64, amount_: u64, color_: b256, value: bool) -> bool;
+}
+
+impl FooCaller for Contract {
+    fn call_foo_contract(gas_: u64, amount_: u64, color_: b256, value: bool) -> bool {
+        let foo_contract = abi(FooContract, 0x7b4837e641d659a0662183f0fdfeca3fb4fa1248d62e4721ff28808bf11bd8c7);
```

Comment:
maybe document what this ID is/comes from

Reply:
It's the `FooContract` address... right there next to itself: 
```
      vv the contract,           vv the address
abi(FooContract, 0x7b4837e641d659a0662183f0fdfeca3fb4fa1248d62e4721ff28808bf11bd8c7)
```

Reply:
But I mean did you choose the address yourself? Is it deterministacally always the same contract address?